.github/actions/check-changes/action.yml

```diff
@@ -0,0 +1,35 @@
+name: Check for changes
+inputs:
+  glob:
+    description: "Glob pattern used to search for changes"
+    required: true
+outputs:
+  changes:
+    description: "Set to true in case there are any changes matching the glob pattern"
+    value: ${{ steps.check.outputs.changes }}
+runs:
+  using: composite
+  steps:
+    - name: Check for changes
+      id: check
+      shell: bash
+      run: |
+        changes=""
+        if [ "${{ github.event_name }}" = "pull_request" ]; then
+          echo "Checking pull request changes between HEAD and origin/${{ github.base_ref }}"
+
+          changes="$(git diff 'origin/${{ github.base_ref }}..HEAD' --name-only '${{ inputs.glob }}')"
+        else
+          echo "Checking changes made by the last (merge) commit"
+
+          changes="$(git diff HEAD^ --name-only '${{ inputs.glob }}')"
+        fi
+
+        if [ "$changes" != "" ]; then
+          echo "Found changes matching the glob pattern:"
+          echo "$changes"
```

Comment:
The `--name-only` makes it reasonable to output. If that becomes too verbose, we can use `git diff --quiet` which just gives a verdict as the exit code.

Reply:
> If have added it explicitly because I remembered back then when adding the `dorny/paths-filter` action I wondered which files caused the pipeline to either run or not and I found it very helpful to have the info.

For sure.

> Also to see if the glob pattern works as expected.

I couldn't quickly find any doc about these glob patterns but it seems that multiple patterns are supported like that:

```
git diff --name-only HEAD^ ':!po/*' ':!doc/*'
```

Reply:
Ah, just saw your update with a link to the doc, thanks!